src/test/java/de/tum/cit/aet/artemis/iris/IrisChatSessionResourceTest.java

```diff
@@ -145,14 +145,25 @@ void getAllSessionsForCourseWithSessions_shouldReturnSessionsWithMessages() thro
         saveChatSessionWithMessages(IrisChatSessionFactory.createLectureSessionForUserWithMessages(lecture, user));
         saveChatSessionWithMessages(IrisChatSessionFactory.createCourseSessionForUserWithMessages(course, user));
         saveChatSessionWithMessages(IrisChatSessionFactory.createTextExerciseSessionForUserWithMessages(textExercise, user));
-        saveChatSessionWithMessages(IrisChatSessionFactory.createProgrammingExerciseChatSessionForUserWithMessages(soloExercise, user));
+        saveChatSessionWithMessages(IrisChatSessionFactory.createProgrammingExerciseChatSessionForUserWithMessages(programmingExercise, user));
 
         List<IrisChatSessionDTO> irisChatSessions = request.getList("/api/iris/chat-history/" + course.getId() + "/sessions", HttpStatus.OK, IrisChatSessionDTO.class);
 
         assertThat(irisChatSessions).hasSize(4);
-        assertThat(irisChatSessions.stream().anyMatch(session -> session.entityId().equals(lecture.getId()))).isTrue();
-        assertThat(irisChatSessions.stream().anyMatch(session -> session.entityId().equals(course.getId()))).isTrue();
-        assertThat(irisChatSessions.stream().anyMatch(session -> session.entityId().equals(textExercise.getId()))).isTrue();
-        assertThat(irisChatSessions.stream().anyMatch(session -> session.entityId().equals(soloExercise.getId()))).isTrue();
+        var lectureSession = irisChatSessions.stream().filter(session -> session.entityId().equals(lecture.getId())).findFirst()
+                .orElseThrow(() -> new AssertionError("Lecture session not found"));
+
+        var courseSession = irisChatSessions.stream().filter(session -> session.entityId().equals(course.getId())).findFirst()
+                .orElseThrow(() -> new AssertionError("Course session not found"));
+
+        var textExerciseSession = irisChatSessions.stream().filter(session -> session.entityId().equals(textExercise.getId())).findFirst()
+                .orElseThrow(() -> new AssertionError("Text exercise session not found"));
+
+        var programmingExerciseSession = irisChatSessions.stream().filter(session -> session.entityId().equals(programmingExercise.getId())).findFirst()
+                .orElseThrow(() -> new AssertionError("Programming exercise session not found"));
```

Comment:
To me this seems to be rather hard to read, wouldn't it be simpler to just add a `isNotNull` assertion?

```suggestion
        IrisChatSessionDTO lectureSession = irisChatSessions.stream().filter(session -> session.entityId().equals(lecture.getId())).findFirst().orElse(null);
        assertThat(lectureSession).isNotNull();

        IrisChatSessionDTO courseSession = irisChatSessions.stream().filter(session -> session.entityId().equals(course.getId())).findFirst().orElse(null);
        assertThat(courseSession).isNotNull();

        IrisChatSessionDTO textExerciseSession = irisChatSessions.stream().filter(session -> session.entityId().equals(textExercise.getId())).findFirst().orElse(null);
        assertThat(textExerciseSession).isNotNull();

        IrisChatSessionDTO programmingExerciseSession = irisChatSessions.stream().filter(session -> session.entityId().equals(programmingExercise.getId())).findFirst().orElse(null);
        assertThat(programmingExerciseSession).isNotNull();
```